lib/matplotlib/axes/_axes.py

```diff
@@ -2419,24 +2421,37 @@ def stem(self, *args, **kwargs):
             basefmt = kwargs.pop('basefmt', args[2])
         except IndexError:
             basefmt = kwargs.pop('basefmt', 'r-')
+        try:
+            orientation = kwargs.pop('orientation', args[3])
+        except IndexError:
+            orientation = kwargs.pop('orientation', 'vertical')
 
         bottom = kwargs.pop('bottom', None)
         label = kwargs.pop('label', None)
-
-        markerline, = self.plot(x, y, markerfmt, label="_nolegend_")
+        if (orientation == "vertical"):
+            markerline, = self.plot(x, y, markerfmt, label="_nolegend_")
+        elif (orientation == "horizontal"):
+            markerline, = self.plot(y, x, markerfmt, label="_nolegend_")
 
         if bottom is None:
             bottom = 0
 
         stemlines = []
         for thisx, thisy in zip(x, y):
-            l, = self.plot([thisx, thisx], [bottom, thisy], linefmt,
+            if (orientation == "vertical"):
+                l, = self.plot([thisx, thisx], [bottom, thisy], linefmt,
+                           label="_nolegend_")
```

Comment:
do you mean something like this?
l, = self.plot([thisx, thisx], [bottom, thisy], linefmt, 
                   label="_nolegend_")